### PR TITLE
fix: 5777 switch organism bug

### PR DESCRIPTION
**Pull request by tihuan**
## Reason for Change

- #5777

## Changes

1. Add a few more design token functions in `frontend/src/common/theme.ts`
2. Add SC Button in `frontend/src/components/common/Button/index.tsx`
3. Update `selectOrganism` reducer to fix bug (same fix as in Seve's PR)
4. Add a new selector `selectHasCustomFiltersOrGenesSelected` to detect when the redux state is dirty and need to pop the confirm modal
5. Add Confirm Dialog (frontend/src/views/WheresMyGeneV2/components/Filters/components/Organism/components/Dialog/index.tsx)
6. Update `Organism/connect.ts` to use a new state `pendingOrganism` as a placeholder, so if the user cancel the confirm dialog, it won't dispatch an action to update the organism in the store
7. BONUS: Extract more test constants `frontend/tests/common/constants.ts`
8. Add 10 e2e tests for this feature

## Testing steps

The confirm modal should show up when the user makes any changes to filters on the left panel, cellTypes, and genes

1. Navigate to rdev: https://pr-6075-frontend.rdev.single-cell.czi.technology/gene-expression
9. Change "color scale" to "Unscaled"
10. Change organism to mouse
11. You should see the new confirm dialog
12. Click cancel
13. Click on organism dropdown again, and human should still be selected
14. Select mouse again and click confirm
15. The filters should be reset and mouse tissues are rendered

You can repeat this test with any filters, cellTypes, genes, and the confirm modal should pop up in all cases!

Screenshot 2023-10-25 at 10 23 34 AM

https://github.com/chanzuckerberg/single-cell-data-portal/assets/6309723/7e9168e7-a1cc-41c9-ae2d-b27012f3c7e0

### CODE REVIEW

**Comment by tihuan** [comment]
Both @signechambers1 and @karenliang0 have reviewed the PR!

**Review by kaloster — APPROVED** [review]
LGTM! 🚀

File: frontend/src/common/theme.ts [hunk 1]
```diff
@@ -270,6 +270,9 @@ export const gray500 = grey500;
 export const grey600 = (props: CommonThemeProps) => getColors(props)?.gray[600];
 export const gray600 = grey600;
 
+export const greyWhite = () => "#fffff";
```
Comment (seve) [inline frontend/src/common/theme.ts:273]: Bug! `#fffff` is an invalid css hex color

Comment (seve) [inline frontend/src/common/theme.ts:273]: Bug! `#fffff` is an invalid css hex color

Comment (seve) [inline frontend/src/common/theme.ts:273]: Bug! `#fffff` is an invalid css hex color

Comment (kaloster) [reply frontend/src/common/theme.ts:273]: 😱

**Pull request merged**